gateway/api/guardrails/guardrails.go

```diff
@@ -86,23 +97,32 @@ func Put(c *gin.Context) {
 		Output:      req.Output,
 		UpdatedAt:   time.Now().UTC(),
 	}
-	err := models.UpdateGuardRailRules(rule)
+
+	log.Infof("Updating guardrail %s with connections: %v", rule.ID, req.ConnectionIDs)
+
+	// Update guardrail and associate connections in a single transaction
+	err := models.UpsertGuardRailRuleWithConnections(rule, req.ConnectionIDs, false)
+
 	switch err {
 	case models.ErrNotFound:
 		c.JSON(http.StatusNotFound, gin.H{"message": err.Error()})
 		return
 	case nil:
+		log.Infof("Guardrail %s updated successfully with %d connection IDs",
```

Comment:
Remove this log statement please, it's not necessary since we're already auditing this endpoint in a middleware